mlir/lib/IR/OperationSupport.cpp

```diff
@@ -768,7 +768,9 @@ OperationEquivalence::isRegionEquivalentTo(Region *lhs, Region *rhs,
       lhs->getNumSuccessors() != rhs->getNumSuccessors() ||
       lhs->getNumOperands() != rhs->getNumOperands() ||
       lhs->getNumResults() != rhs->getNumResults() ||
-      lhs->hashProperties() != rhs->hashProperties())
+      lhs->hashProperties() != rhs->hashProperties() ||
+      !lhs->getName().compareOpProperties(lhs->getPropertiesStorage(),
+                                        rhs->getPropertiesStorage()))
```

Comment:
I'm actually wondering if hashing isn't even more costly than the comparison somehow, wonder if we should drop the hash comparison here?

Reply:
You are probably right for most cases. And for the few exceptions, where hashing actually yields an advantage, it can still be considered when overloading the equality operator for the specific type of properties.